### PR TITLE
Add a note and diagram to clarify legacy mouse pointer behavior.

### LINKED ISSUE #454: Order of pointerover/enter/move and corresponding mouse events is different on browsers
`data:text/html,<input id=input><script>input.onpointerenter = input.onmouseenter = input.onpointerleave = input.onmouseleave = input.onpointerover = input.onmouseover = input.onpointerout = input.onmouseout = input.onpointermove = input.onmousemove = ev =>console.log(ev.type)</script>`

When hovering over the input, the console output is:

Chrome: pointerover->pointerenter->mouseover->mouseenter->pointermove->mousemove->pointerout->pointerleave->mouseout->mouseleave
Firefox: pointerover->pointerenter->pointermove->mouseover->mouseenter->mousemove->pointerout->pointerleave->mouseout->mouseleave
Safari: pointerover->mouseover->pointerenter->mouseenter->pointermove->mousemove->pointerout->mouseout->pointerleave->mouseleave

Somehow the timing that pointermove and mousemove appear is different.

* The pointer/mousemove happens immediately on pointerenter on all browsers
* Events are somehow grouped on Firefox and Chrome (pointerover+pointerenter always happen together without a compatibility event between it, for example)
* And the grouping behaviors are different.

I'm not quite sure this grouping is specced. https://w3c.github.io/pointerevents/#mapping-for-devices-that-support-hover looks like it doesn't even give the mapping for over/enter/out/leave.

### PR DESCRIPTION
**Pull request by mustaqahmed**
This shows the effective position of legacy mouse pointer when two primary pointers are active at the same time, and points out the disconnection between `pointerout` and `mouseout` events.

Closes #454.

***
<a href="https://pr-preview.s3.amazonaws.com/mustaqahmed/pointerevents/pull/458.html" title="Last updated on Oct 26, 2022, 7:34 PM UTC (5f4cb35)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/pointerevents/458/1074161...mustaqahmed:5f4cb35.html" title="Last updated on Oct 26, 2022, 7:34 PM UTC (5f4cb35)">Diff</a>

### CODE REVIEW

**Comment by mustaqahmed** [comment]
@flackr and @patrickhlauke PTAL.  I will fine-tune on the animated GIF once the text seems okay.

File: index.html [hunk 1]
```diff
@@ -1204,6 +1204,32 @@ <h2><dfn>Tracking the effective position of the legacy mouse pointer</dfn></h2>
...
+              <code>mouseout</code>, <code>mouseenter</code> and <code>mouseleave</code>).  This is because the
+              mouse transition events must simulate the movement of a <emph>single</emph> legacy mouse.  The
+              following animation shows such a situation, where the orange pointer represents the <a>effective position
+              of the legacy mouse pointer</a>.
+
+              <figure id="legacy-pointer-transition">
+                <img src="images/legacy-pointer-animation.webm"
+                     alt="legacy mouse pointer animation">
+                <figcaption>
+                  Simultaneous mouse and touch pointers causing legacy mouse to move between the two pointers.
```
Comment (flackr) [inline index.html]: This could really use an explanation of the orange pointer (being the simulated pointer) and the white cursor and pointer.

Comment (flackr) [reply index.html]: Actually I suppose this is mentioned above, but it wouldn't hurt to clarify, e.g.

 Simultaneous mouse (white cursor) and touch pointer (white pointer) causing legacy mouse (orange cursor) to move between the two pointers.

Comment (mustaqahmed) [reply index.html]: Done.

**Comment by patrickhlauke** [comment]
Did some tweaking/rewording

File: index.html [hunk 2]
```diff
@@ -1196,14 +1196,33 @@ <h1><dfn data-lt="compatibility mouse events">Compatibility mapping with mouse e
...
                 <li>If |T| and current <a data-lt="effective position of the legacy mouse pointer">effective legacy mouse pointer position</a> are both unset or they are equal, terminate these steps.</li>
                 <li>Dispatch <code>mouseover</code>, <code>mouseout</code>, <code>mouseenter</code> and <code>mouseleave</code> events as per [[UIEVENTS]] for a mouse moving from the current <a data-lt="effective position of the legacy mouse pointer">effective legacy mouse pointer position</a> to |T|. Consider an unset value of either current <a data-lt="effective position of the legacy mouse pointer">effective legacy mouse pointer position</a> or |T| as an out-of-window mouse position.</li>
                 <li>Set <a data-lt="effective position of the legacy mouse pointer">effective legacy mouse pointer position</a> to |T|.</li>
             </ol>
+            <div class="note">
+              <p>The following animation illustrates how a user agent can dispatch compatibility mouse events that reconcile two active primary pointers with an idealised legacy mouse input.</p>
```
Comment (mustaqahmed) [inline index.html]: I think we need to highlight why the spec deliberately leaves out a direct mapping of legacy transition events (over/enter/out/leave events).  The last line in the spec issue reflects a similar confusion we saw before (perhaps internally in Chrome, can't recall).

I am working on a fix now.

Comment (mustaqahmed) [reply index.html]: Done, please take a look.

File: index.html [hunk 3]
```diff
@@ -1196,7 +1196,7 @@ <h1><dfn data-lt="compatibility mouse events">Compatibility mapping with mouse e
...
+            <p>While only <a>primary pointers</a> can produce compatibility mouse events, <a href="#multiple-primary-pointers">multiple primary pointers</a> can be active simultaneously, each producing its own compatibility mouse events. For compatibility with scripts relying on MouseEvents, the mouse transition events (<code>mouseover</code>, <code>mouseout</code>, <code>mouseenter</code> and <code>mouseleave</code>) SHOULD simulate the movement of a <em>single</em> legacy mouse input. This means that the entry/exit state for every event target is valid, in accordance with [[UIEVENTS]]. Users agents SHOULD guarantee this by maintaining the <dfn>effective position of the legacy mouse pointer</dfn> in the document as follows.</p>
```
Comment (patrickhlauke) [inline index.html:1199]: Good catch listing the actual `mouse*` events here, rather than `pointer*` ones

File: index.html [hunk 4]
```diff
@@ -1205,23 +1205,28 @@ <h2><dfn>Tracking the effective position of the legacy mouse pointer</dfn></h2>
                 <li>Set <a data-lt="effective position of the legacy mouse pointer">effective legacy mouse pointer position</a> to |T|.</li>
             </ol>
             <div class="note">
-              <p>The following animation illustrates how a user agent can dispatch compatibility mouse events that reconcile two active primary pointers with an idealised legacy mouse input.</p>
+              <p> The <a>effective position of the legacy mouse pointer</a> models the fact that we cannot always have a direct mapping
+                from pointer transition events (i.e., <code>pointerover</code>, <code>pointerout</code>, <code>pointerenter</code>
+                and <code>pointerleave</code>) to corresponding legacy mouse transition events (i.e., <code>mouseover</code>,
+                <code>mouseout</code>, <code>mouseenter</code> and <code>mouseleave</code>).  The following animation illustrates a case
+                where a user agent needs to dispatch more legacy mouse transition events than pointer transition events to be able to
+                reconcile two primary pointers using a single legacy mouse input.</p>
```
Comment (patrickhlauke) [inline index.html:1213]: Looking good, thanks

**Review by patrickhlauke — APPROVED** [review]

File: index.html (hunk 2 above)
Comment (patrickhlauke) [reply index.html]: Thanks...these changes look great to me

**Pull request merged**